src/main/java/gg/solarmc/kitpvp/kill/KillDataHandler.java

```diff
@@ -2,55 +2,116 @@
 
 import gg.solarmc.kitpvp.KitpvpConfig;
 import gg.solarmc.loader.DataCenter;
-import gg.solarmc.loader.credits.CreditsKey;
+import gg.solarmc.loader.SolarPlayer;
 import gg.solarmc.loader.kitpvp.KitPvpKey;
-import org.bukkit.entity.Player;
 import space.arim.omnibus.util.concurrent.CentralisedFuture;
 
-import java.math.BigDecimal;
-import java.util.Collection;
+import java.util.Optional;
 import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
 
 public class KillDataHandler {
 
+    private final DataCenter center;
+    private final KitpvpConfig config;
+
+
     public KillDataHandler(DataCenter center, KitpvpConfig config) {
         this.center = center;
         this.config = config;
     }
 
-    private final DataCenter center;
-    private final KitpvpConfig config;
+    CentralisedFuture<?> handleKill(UUID killerId, UUID killedId, Set<UUID> assisters) {
+
+        CentralisedFuture<Optional<SolarPlayer>> killedFuture = center.lookupPlayer(killedId);
+        CentralisedFuture<Optional<SolarPlayer>> killerFuture = center.lookupPlayer(killerId);
+
+        CentralisedFuture<Void> combine = (CentralisedFuture<Void>) CentralisedFuture.allOf(killedFuture,killerFuture);
+        //can you fix this on centralized future (casting) ? I'm sure as hell fixing this for taskfuture >:)
```

Comment:
Before you delve into completable future operations, you should consider the intent of your code particularly what you are trying to achieve.

In this case, the killer and the killed player will always be online. That means you don't need to perform an offline player lookup (which you are doing here), and thus you don't need any future solar players nor any optionals.

Player#getSolarPlayer is what you want to use here. I suggest changing the parameter types to Player instead of UUID.

Reply:
Sounds good.